src/components/FormDay.js

```diff
@@ -0,0 +1,154 @@
+import styled from 'styled-components';
+import { useForm } from 'react-hook-form';
+import getDates from './hooks/getDates.js';
+import getDisplayDate from './hooks/getDisplayDate.js';
+import { useImmer } from 'use-immer';
+
+export default function FormDay({
+  currentDestination,
+  formName,
+  buttonName,
+  setNewDay,
+}) {
+  console.log(currentDestination);
+  const startDate = new Date(currentDestination[0].startDate);
+  const endDate = new Date(currentDestination[0].endDate);
+  const dates = getDates(startDate, endDate);
+  const [formLocations, updateFormLocations] = useImmer(
+    currentDestination[0].locations.map(loc => ({
+      location: loc,
+      isChecked: false,
+    }))
+  );
+  const plannedDays = currentDestination[0].routes.map(day =>
+    new Date(day.date).getTime()
+  );
+  const filteredDates = dates.filter(
+    day => !plannedDays.includes(day.getTime())
+  );
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    mode: 'all',
+  });
+
+  function handleToggle(e) {
+    updateFormLocations(draft => {
+      const currentLocation = draft.find(
+        locat => locat.location === e.target.value
+      );
+      currentLocation.isChecked = !currentLocation.isChecked;
+    });
+  }
+
+  const onSubmit = data => {
+    setNewDay(data, formLocations);
+  };
+  return (
+    <>
+      <FormContainer
+        aria-label={formName}
+        id="newDayForm"
+        autoComplete="off"
+        onSubmit={handleSubmit(onSubmit)}
+      >
+        <Wrapper>
+          <LabelHeader htmlFor="select date">Date</LabelHeader>
+          <StyledSelect
+            name="date"
+            id="select date"
+            autoFocus
+            {...register('date', { required: 'Please select a date' })}
+          >
+            <option value={''}>--Select a date--</option>
+            {filteredDates.map((date, index) => (
+              <option key={index} value={date}>
+                {getDisplayDate(date)}
+              </option>
+            ))}
+          </StyledSelect>
+          {errors?.date && <ErrorMessage>{errors?.date.message}</ErrorMessage>}
+        </Wrapper>
+        <Wrapper>
+          {formLocations?.map((loc, index) => (
+            <LocationButton
+              key={index}
+              type="button"
+              value={loc.location}
+              onClick={e => handleToggle(e)}
+              className={loc.isChecked ? 'active' : ''}
+            >
+              {loc.location}
+            </LocationButton>
+          ))}
+        </Wrapper>
+        <CreateButton>{buttonName}</CreateButton>
+      </FormContainer>
+    </>
+  );
+}
+
+const FormContainer = styled.form`
+  display: grid;
+  grid-template-rows: repeat(6, auto);
+  gap: 25px;
+  max-width: 400px;
+  width: 100%;
+  margin-left: auto;
+  margin-right: auto;
+`;
+
+const Wrapper = styled.div`
+  max-width: 400px;
```

Comment:
not sure if u actually need two times `max-width: 400px`...if it should be more in general you could also think about to place it in the the app or even globalstyles